site/en/docs/extensions/mv3/getstarted/development-basics/index.md

```diff
@@ -0,0 +1,271 @@
+    "default_icon": "hello_extensions.png"
+ }
+}
+```
+
+The first fields contain the metadata of the extension. Under the `"action"` key, the extension
+declares the popup and action icon. You can download the icon [here][hello-icon]. 
+
+For the popup, create a file named `hello.html`, and add the following code:
+
+{% Label %}hello.html{% endLabel %}
+
+``` html
+<html>
+  <body>
+    <h1>Hello Extensions</h1>
+  </body>
+</html>
+```
+
+The extension now displays a popup when the extension action (toolbar icon) is clicked. Let's test
+it on Chrome by loading it locally. Ensure all files are saved.
+
+## Loading an unpacked extension {: #load-unpacked }
+
+To load an unpacked extension in developer mode, follow these steps:
+
+1. Go to the Extension Management page by navigating to `chrome://extensions`.
+    - Alternatively, click on the Extensions menu puzzle button and select **Manage Extensions** at the bottom of the menu.
+    - Or, click on the Chrome menu, hovering over **More Tools,** then select **Extensions**.
+2. Enable Developer Mode by clicking the toggle switch next to **Developer mode**.
+3. Click the **Load unpacked** button and select the extension directory.
+    <figure>
+    {% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/BzVElZpUtNE4dueVPSp3.png", alt="Chrome Extension Management page", 
+    width="400", height="183",  class="screenshot" %}
+
+      <figcaption>
+      Chrome Management (chrome://extensions)
+      </figcaption>
+    </figure>
+
+Ta-da! The extension has been successfully installed. Because no extension icons were included in
+the manifest, a generic icon will be created for the extension.
+
+## Pinning the extension {: #pin }
+
+By default, when you load your extension locally, it will appear in the **extensions menu** {% Img
+src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/f5i7AgfauCfoQJxnn3kU.png", alt="Puzzle", width="24",
+height="24" %}. Pin your extension to the toolbar to quickly access your extension during
+development.
+
+<figure>
+{% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/LahIugYHQW3QpHM0z1qZ.png", 
+alt="Pinning the extension", width="358", height="248", class="screenshot" %}
+  <figcaption>
+  Pinning the extension
+  </figcaption>
+</figure>
+
+Click on the extension’s action (toolbar icon); you should see a popup.
+
+<figure>
+{% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/xjKRmWMgwMm7Kdf72bkj.png", 
+alt="hello world extension", width="206", height="130", class="screenshot" %}
+  <figcaption>
+  Hello World extension
+  </figcaption>
+</figure>
+
+## Reloading the extension {: #reload }
+
+Let’s go back to the code and change the content of the `H1` element in your HTML file.
+
+{% Label %}hello.html{% endLabel %}
+
+```html
+<html>
+ <body>
+   <h1>Hello Extensions of the world! 👋</h1>
+ </body>
+</html>
+```
+
+After saving the file, go to the Extension Management page. You can see the changes made by clicking
+on the refresh icon next to the **on/off** toggle:
```

Comment:
You are right! I struggled to keep the example simple, plus at the same time, introduce development workflow concepts. You got me there! I was trying to avoid adding a service worker in this example. 

I'd be open to conversing about this further.

Reply:
### Here's the breakdown

**Do not require extension reload**
- Popup HTML
- Scripts inside the HTML
- Options HTML page
- Other extension HTML pages.

**Require extension reload**
- Content scripts declared in the manifest (extension reload & host page refresh).
- Manifest
- Service worker

### Moving forward

- Modify the manifest name instead of the popup <H1> tag.
- Add a details component including which components require the extension to be refreshed.

Reply:
[See preview](https://deploy-preview-3328--developer-chrome-com.netlify.app/docs/extensions/mv3/getstarted/development-basics/#reload)